crates/router/tests/connectors/stax.rs

```diff
@@ -32,20 +34,81 @@ impl utils::Connector for StaxTest {
 
 static CONNECTOR: StaxTest = StaxTest {};
 
-fn get_default_payment_info() -> Option<utils::PaymentInfo> {
-    None
+fn get_default_payment_info(
+    connector_customer: Option<String>,
+    payment_method_token: Option<String>,
+) -> Option<utils::PaymentInfo> {
+    Some(utils::PaymentInfo {
+        address: None,
+        auth_type: None,
+        access_token: None,
+        connector_meta_data: None,
+        return_url: None,
+        connector_customer,
+        payment_method_token,
+        payout_method_data: None,
+        currency: None,
+        country: None,
+    })
+}
+
+fn customer_details() -> Option<types::ConnectorCustomerData> {
+    Some(types::ConnectorCustomerData {
+        ..utils::CustomerType::default().0
+    })
+}
+
+fn token_details() -> Option<types::PaymentMethodTokenizationData> {
+    Some(types::PaymentMethodTokenizationData {
+        payment_method_data: types::api::PaymentMethodData::Card(api::Card {
+            card_number: cards::CardNumber::from_str("4111111111111111").unwrap(),
+            card_exp_month: Secret::new("04".to_string()),
+            card_exp_year: Secret::new("2027".to_string()),
+            card_cvc: Secret::new("123".to_string()),
+            ..utils::CCardType::default().0
+        }),
+        browser_info: None,
+    })
 }
 
 fn payment_method_details() -> Option<types::PaymentsAuthorizeData> {
-    None
+    Some(types::PaymentsAuthorizeData {
+        ..utils::PaymentAuthorizeType::default().0
+    })
 }
 
 // Cards Positive Tests
 // Creates a payment using the manual capture flow (Non 3DS).
 #[actix_web::test]
 async fn should_only_authorize_payment() {
+    let customer_response = CONNECTOR
+        .create_connector_customer(customer_details(), get_default_payment_info(None, None))
+        .await
+        .expect("Authorize payment response");
+    let connector_customer_id = match customer_response.response.unwrap() {
+        PaymentsResponseData::ConnectorCustomerResponse {
+            connector_customer_id,
+        } => Some(connector_customer_id),
+        _ => None,
+    };
+
+    let token_response = CONNECTOR
+        .create_connector_pm_token(
+            token_details(),
+            get_default_payment_info(connector_customer_id, None),
+        )
+        .await
+        .expect("Authorize payment response");
+    let payment_method_token = match token_response.response.unwrap() {
+        PaymentsResponseData::TokenizationResponse { token } => Some(token),
+        _ => None,
+    };
```

Comment:
Move this to common method as this same piece of code is being used in all your test